File: src/main/java/online/maestoso/soulforged/command/Commands.java
package online.maestoso.soulforged.command;

import net.fabricmc.fabric.api.command.v1.CommandRegistrationCallback;

public class Commands {
    public static void register() {
        CommandRegistrationCallback.EVENT.register(((dispatcher, dedicated) -> {
            GiveToolCommand.register(dispatcher);
        }));
    }
}

File: src/main/kotlin/studio/soulforged/soulforged/command/Commands.kt
package studio.soulforged.soulforged.command

import net.minecraft.command.argument.SingletonArgumentInfo
import org.quiltmc.qsl.command.api.CommandRegistrationCallback
import org.quiltmc.qsl.command.api.ServerArgumentType
import studio.soulforged.soulforged.Soulforged.id

object Commands {
    internal fun register() {
        ServerArgumentType.register("tool_type".id(), ToolTypeArgumentType::class.java,
            SingletonArgumentInfo.contextFree { ToolTypeArgumentType() }) { ToolTypeArgumentType() }
        ServerArgumentType.register("material".id(), MaterialArgumentType::class.java,
            SingletonArgumentInfo.contextFree { MaterialArgumentType() }) { MaterialArgumentType() }
        CommandRegistrationCallback.EVENT.register(CommandRegistrationCallback { dispatcher, _, _ ->
            GiveToolCommand.register(dispatcher)
        })
    }
}